oauthproxy.go

```diff
@@ -1110,6 +1119,19 @@ func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Req
 	}
 }
 
+// stripAuthHeaders removes Auth headers for whitelisted routes from skipAuthRegex
+func (p *OAuthProxy) stripAuthHeaders(req *http.Request) {
+	if p.PassUserHeaders {
+		req.Header.Del("X-Forwarded-User")
+		req.Header.Del("X-Forwarded-Email")
+		req.Header.Del("X-Forwarded-Preferred-Username")
+	}
+
+	if p.PassAccessToken {
```

Comment:
What about other pass flags such as pass authorization?

Reply:
I leaned toward only messing with `X-Forwarded-*` flavored headers that are potentially semi-unique to our proxy.

I thought about the other ones, but leaned against stripping the `Authorization` header since that is a real spec HTTP header that people are potentially using -- my opinion is very sway-able, I almost implemented it that way at first 😄 

Plus -- there's the added benefit that IDTokens in Authorization headers are much less spoofable (but alot of backends are potentially just taking claims and not validating the signature... so maybe that point is moot)

Reply:
Actually yeah -- I think we should leave it in place and not touch the authorization header.  I can think of examples where I've seen users have a whitelist to a route consumed by a public SaaS product and they used there own token auth scheme in code to support the vendor's capabilities.

Reply:
Where we have previously implemented this kind of thing, we have stripped any header that is configured to be passed.

Basically we concluded that if we are proxying and setting a header, only the proxy should be setting those headers to the upstream. Otherwise you'll end up with a mixture of sources which we deemed to be insecure. I'm willing to be convinced otherwise though?

Reply:
> Actually yeah -- I think we should leave it in place and not touch the authorization header.  I can think of examples where I've seen users have a whitelist to a route consumed by a public SaaS product and they used their own token auth scheme in code to support the vendor's capabilities.

Hadn't really thought about this as a use case before, but you're right, I think people are doing this. What's to say they aren't doing it for the X forwarded headers as well though? 🤔

Reply:
I guess we have a flag added in this PR to allow users to not strip headers if they really want their own bearer headers in the whitelisted paths.

I can make the tweak 👍 

I think this would possibly result in most cases stripping the `Authorization` header since default settings add in the `User` as a `Basic` in there?

Reply:
Yes I think it would. Though I think there are several options that might trigger authorization headers to be set and therefore several options that would trigger this removal

Reply:
> > Actually yeah -- I think we should leave it in place and not touch the authorization header.  I can think of examples where I've seen users have a whitelist to a route consumed by a public SaaS product and they used their own token auth scheme in code to support the vendor's capabilities.
> 
> Hadn't really thought about this as a use case before, but you're right, I think people are doing this. What's to say they aren't doing it for the X forwarded headers as well though? 🤔

They could totally be doing it in X-Forwarded headers, I agree (hence I added the flag to disabled this).

But that feels much more hacky and insecure than allowing in alternative Authorization headers -- much easier to spoof `X-Forwarded-User: Nick` than a properly signed IDToken or a legitimate Authorization header that a backend validates (Basic or Bearer).

I'd hope users didn't do that -- but people do hacky, insecure stuff when deadlines approach 😄

Reply:
I wonder if it makes sense to flip the default value for `--pass-basic-auth` to false? Then the default proxy case wouldn't touch the `authorization` header and most implementations wouldn't strip it?

I wonder how many users use that bare `basic user:` authorization header to get the user instead of the X headers would be affected by that flip.

I was surprised by the unintentional default value of that being true testing a CLI PKSE app with a dummy oauth2-proxy + httpbin using the `--skip-jwt-bearer-tokens` earlier today.  The default `--pass-basic-auth` replaced the IDToken I was curling in the bearer header with the basic user

Reply:
> I wonder if it makes sense to flip the default value for --pass-basic-auth to false? Then the default proxy case wouldn't touch the authorization header and most implementations wouldn't strip it?

I guess I would be expecting the proxy not to inject anything unless I explicitly ask it to, so that seems reasonable to me